packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/auth/ApiKeyAuth.kt

```diff
@@ -0,0 +1,69 @@
+package io.realm.kotlin.mongodb.auth
+
+import io.realm.kotlin.mongodb.App
+import io.realm.kotlin.mongodb.User
+import io.realm.kotlin.types.ObjectId
+
+/**
+ * Exposes functionality for a user to manage API keys under their control.
+ */
+public interface ApiKeyAuth {
+
+    /**
+     * The User that this instance in associated with.
+     */
+    public val user: User
+
+    /**
+     * The App that this instance in associated with.
+     */
+    public val app: App
+
+    /**
+     * Creates a user API key that can be used to authenticate as the user.
+     * The value of the key must be persisted at this time as this is the only time it is visible.
+     * The key is enabled when created. It can be disabled by calling the disable method.
+     *
+     * @param name the name of the key
+     * @throws IllegalArgumentException if an invalid name for the key is sent to the server.
+     * @return the new API key for the user.
+     */
```

Comment:
```suggestion
    /**
     * Creates an [ApiKey] that can be used to authenticate as the user.
     * The value of the key must be persisted at this time as this is the only time it is visible.
     * The key is enabled by default when created. It can be disabled by calling [disable].
     *
     * @param name the name of the key.
     * @throws `IllegalArgumentException` if an invalid name for the key is sent to the server.
     * @return the new API key for the user.
     */
```